src/injected/textSelectorEngine.ts

```diff
@@ -79,5 +57,54 @@ function createMatcher(selector: string): Matcher {
     return text => re.test(text);
   }
   selector = selector.trim().toLowerCase();
-  return text => text.trim().toLowerCase() === selector;
+  return text => text.toLowerCase().includes(selector);
+}
+
+function queryInternal(root: SelectorRoot, matcher: Matcher): Element | undefined {
+  const document = root instanceof Document ? root : root.ownerDocument;
+  if (!document)
```

Comment:
```suggestion
  assert(document);
```

This is an assert right? ownerDocument should be never null unless root is a document.

Reply:
That's right! We don't have asserts in the injected code though, so I am not sure how to change this.

Reply:
just `document!` then? The if there makes me try to reason about the null case and I go down a bad path.